// cloudinary-react-native/lib/module/widgets/video/layer/components/BottomControls.js
import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { ButtonPosition } from '../types';
import { getResponsiveStyles } from '../styles';
import { ICON_SIZES } from '../constants';
import { Seekbar } from './Seekbar';
import { PlaybackSpeedButton } from './PlaybackSpeedButton';
import { SubtitlesButton } from './SubtitlesButton';
import { QualityButton } from './QualityButton';
export const BottomControls = ({
  status,
  onPlayPause,
  onMuteToggle,
  formatTime,
  getProgress,
  getCurrentPosition,
  seekBarRef,
  panResponder,
  backButtonPosition,
  shareButtonPosition,
  isLandscape = false,
  seekbar = {},
  playbackSpeed,
  currentPlaybackSpeed,
  onPlaybackSpeedChange,
  isSpeedMenuVisible,
  onToggleSpeedMenu,
  subtitles,
  currentSubtitle,
  onSubtitleChange,
  isSubtitlesMenuVisible,
  onToggleSubtitlesMenu,
  quality,
  currentQuality,
  onQualityChange,
  isQualityMenuVisible,
  onToggleQualityMenu
}) => {
  const responsiveStyles = getResponsiveStyles(isLandscape);
  const progress = getProgress();
  const currentPosition = getCurrentPosition();

  // Check if we need to leave space for SE positioned buttons
  const hasSEButton = backButtonPosition === ButtonPosition.SE || shareButtonPosition === ButtonPosition.SE;
  return /*#__PURE__*/React.createElement(View, {
    style: [responsiveStyles.bottomControlsBar, hasSEButton && {
      paddingRight: 80
    } // Create space for SE button
    ]
  }, /*#__PURE__*/React.createElement(View, {
    style: responsiveStyles.bottomLeftControls
  }, /*#__PURE__*/React.createElement(TouchableOpacity, {
    style: responsiveStyles.playPauseButton,
    onPress: onPlayPause
  }, /*#__PURE__*/React.createElement(Ionicons, {
    name: status !== null && status !== void 0 && status.isPlaying ? 'pause' : 'play',
    size: ICON_SIZES.bottomPlayPause,
    color: "white",
    style: responsiveStyles.playPauseIcon
  })), /*#__PURE__*/React.createElement(Seekbar, {
    progress: progress,
    currentPosition: currentPosition,
    status: status,
    formatTime: formatTime,
    seekBarRef: seekBarRef,
    panResponder: panResponder,
    isLandscape: isLandscape,
    seekBar: seekbar
  })), /*#__PURE__*/React.createElement(View, {
    style: responsiveStyles.bottomRightControls
  }, /*#__PURE__*/React.createElement(QualityButton, {
    quality: quality,
    currentQuality: currentQuality,
    onQualityChange: onQualityChange,
    isLandscape: isLandscape,
    isMenuVisible: isQualityMenuVisible,
    onToggleMenu: onToggleQualityMenu
  }), /*#__PURE__*/React.createElement(SubtitlesButton, {
    subtitles: subtitles,
    currentSubtitle: currentSubtitle,
    onSubtitleChange: onSubtitleChange,
    isLandscape: isLandscape,
    isMenuVisible: isSubtitlesMenuVisible,
    onToggleMenu: onToggleSubtitlesMenu
  }), /*#__PURE__*/React.createElement(PlaybackSpeedButton, {
    playbackSpeed: playbackSpeed,
    currentSpeed: currentPlaybackSpeed,
    onSpeedChange: onPlaybackSpeedChange,
    isLandscape: isLandscape,
    isMenuVisible: isSpeedMenuVisible,
    onToggleMenu: onToggleSpeedMenu
  }), /*#__PURE__*/React.createElement(TouchableOpacity, {
    style: responsiveStyles.volumeButton,
    onPress: onMuteToggle
  }, /*#__PURE__*/React.createElement(Ionicons, {
    name: status !== null && status !== void 0 && status.isMuted ? 'volume-mute' : 'volume-high',
    size: ICON_SIZES.bottomVolume,
    color: "white",
    style: responsiveStyles.volumeIcon
  }))));
};
