// cloudinary-react-native/lib/module/widgets/video/layer/components/BottomButtonBar.js
import React from 'react';
import { View, TouchableOpacity, Text } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
export function BottomButtonBar({
  isControlsVisible,
  bottomButtonBar
}) {
  var _bottomButtonBar$styl, _bottomButtonBar$styl2, _bottomButtonBar$styl3, _bottomButtonBar$styl4, _bottomButtonBar$styl5, _bottomButtonBar$styl6, _bottomButtonBar$styl7;
  if (!isControlsVisible || !(bottomButtonBar !== null && bottomButtonBar !== void 0 && bottomButtonBar.enabled)) return null;
  return /*#__PURE__*/React.createElement(View, {
    style: [{
      position: 'absolute',
      bottom: (() => {
        // Position button bar below the seekbar (closer to screen bottom)
        // Use a small bottom value to place it below the seekbar
        const spacingFromBottom = 0;
        return spacingFromBottom;
      })(),
      left: ((_bottomButtonBar$styl = bottomButtonBar.style) === null || _bottomButtonBar$styl === void 0 ? void 0 : _bottomButtonBar$styl.marginHorizontal) || 20,
      right: ((_bottomButtonBar$styl2 = bottomButtonBar.style) === null || _bottomButtonBar$styl2 === void 0 ? void 0 : _bottomButtonBar$styl2.marginHorizontal) || 20,
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
      zIndex: 1,
      // Lower than seekbar and bottom controls
      backgroundColor: ((_bottomButtonBar$styl3 = bottomButtonBar.style) === null || _bottomButtonBar$styl3 === void 0 ? void 0 : _bottomButtonBar$styl3.backgroundColor) || 'rgba(0,0,0,0.7)',
      borderRadius: ((_bottomButtonBar$styl4 = bottomButtonBar.style) === null || _bottomButtonBar$styl4 === void 0 ? void 0 : _bottomButtonBar$styl4.borderRadius) || 20,
      paddingHorizontal: ((_bottomButtonBar$styl5 = bottomButtonBar.style) === null || _bottomButtonBar$styl5 === void 0 ? void 0 : _bottomButtonBar$styl5.paddingHorizontal) || 16,
      paddingVertical: ((_bottomButtonBar$styl6 = bottomButtonBar.style) === null || _bottomButtonBar$styl6 === void 0 ? void 0 : _bottomButtonBar$styl6.paddingVertical) || 8,
      marginBottom: ((_bottomButtonBar$styl7 = bottomButtonBar.style) === null || _bottomButtonBar$styl7 === void 0 ? void 0 : _bottomButtonBar$styl7.marginBottom) || 0
    }]
  }, bottomButtonBar.buttons.map((button, index) => /*#__PURE__*/React.createElement(TouchableOpacity, {
    key: `bottom-bar-${index}`,
    style: {
      marginHorizontal: 16,
      paddingVertical: 8,
      paddingHorizontal: 8,
      backgroundColor: button.backgroundColor || 'transparent',
      borderRadius: button.backgroundColor ? 15 : 0,
      flexDirection: 'row',
      alignItems: 'center'
    },
    onPress: button.onPress || (() => {})
  }, /*#__PURE__*/React.createElement(Ionicons, {
    name: button.icon,
    size: button.size || 20,
    color: button.color || 'white'
  }), button.text && /*#__PURE__*/React.createElement(Text, {
    style: {
      color: button.textColor || button.color || 'white',
      fontSize: 14,
      fontWeight: '500',
      marginLeft: 6
    }
  }, button.text))));
}
