// cloudinary-react-native/lib/module/widgets/video/layer/components/QualityButton.js
import React from 'react';
import { View, TouchableOpacity, Text, Modal, StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { getResponsiveStyles } from '../styles';
import { ICON_SIZES } from '../constants';
export const QualityButton = ({
  quality,
  currentQuality = 'auto',
  onQualityChange,
  isLandscape = false,
  isMenuVisible = false,
  onToggleMenu
}) => {
  var _qualityOptions$find;
  const responsiveStyles = getResponsiveStyles(isLandscape);

  // Don't render if quality selection is not enabled or no options available
  if (!(quality !== null && quality !== void 0 && quality.enabled)) {
    return null;
  }
  const qualityOptions = quality.qualities || [];

  // Don't render if no quality options available
  if (qualityOptions.length === 0) {
    return null;
  }
  const currentQualityLabel = ((_qualityOptions$find = qualityOptions.find(option => option.value === currentQuality)) === null || _qualityOptions$find === void 0 ? void 0 : _qualityOptions$find.label) || 'Auto';
  const handleQualitySelect = qualityValue => {
    onQualityChange === null || onQualityChange === void 0 || onQualityChange(qualityValue);
    onToggleMenu === null || onToggleMenu === void 0 || onToggleMenu(); // Close the menu
  };
  const toggleMenu = () => {
    onToggleMenu === null || onToggleMenu === void 0 || onToggleMenu();
  };
  return /*#__PURE__*/React.createElement(View, {
    style: styles.container
  }, /*#__PURE__*/React.createElement(TouchableOpacity, {
    style: [responsiveStyles.volumeButton, styles.qualityButton],
    onPress: toggleMenu
  }, /*#__PURE__*/React.createElement(Ionicons, {
    name: "cog-outline",
    size: ICON_SIZES.bottomVolume,
    color: "white",
    style: responsiveStyles.volumeIcon
  })), /*#__PURE__*/React.createElement(Modal, {
    transparent: true,
    visible: isMenuVisible,
    animationType: "fade",
    onRequestClose: onToggleMenu
  }, /*#__PURE__*/React.createElement(TouchableOpacity, {
    style: styles.modalOverlay,
    activeOpacity: 1,
    onPress: onToggleMenu
  }, /*#__PURE__*/React.createElement(View, {
    style: [styles.qualityMenu, isLandscape && styles.qualityMenuLandscape]
  }, /*#__PURE__*/React.createElement(View, {
    style: styles.qualityMenuHeader
  }, /*#__PURE__*/React.createElement(Text, {
    style: styles.qualityMenuTitle
  }, "Quality")), qualityOptions.map(option => /*#__PURE__*/React.createElement(TouchableOpacity, {
    key: option.value,
    style: [styles.qualityOption, currentQuality === option.value && styles.qualityOptionSelected],
    onPress: () => handleQualitySelect(option.value)
  }, /*#__PURE__*/React.createElement(View, {
    style: styles.qualityOptionContent
  }, /*#__PURE__*/React.createElement(Text, {
    style: [styles.qualityOptionText, currentQuality === option.value && styles.qualityOptionTextSelected]
  }, option.label), option.resolution && option.value !== 'auto' && /*#__PURE__*/React.createElement(Text, {
    style: styles.qualityOptionSubtext
  }, option.resolution)), currentQuality === option.value && /*#__PURE__*/React.createElement(Ionicons, {
    name: "checkmark",
    size: 20,
    color: "#007AFF"
  })))))));
};
const styles = StyleSheet.create({
  container: {
    position: 'relative'
  },
  qualityButton: {
    marginRight: 8
  },
  modalOverlay: {
    flex: 1,
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
    justifyContent: 'center',
    alignItems: 'center'
  },
  qualityMenu: {
    backgroundColor: 'rgba(20, 20, 20, 0.95)',
    borderRadius: 12,
    padding: 4,
    minWidth: 200,
    maxWidth: 300,
    maxHeight: '70%',
    marginBottom: 100
  },
  qualityMenuLandscape: {
    marginBottom: 50,
    minWidth: 220
  },
  qualityMenuHeader: {
    borderBottomWidth: 1,
    borderBottomColor: 'rgba(255, 255, 255, 0.1)',
    paddingVertical: 12,
    paddingHorizontal: 16
  },
  qualityMenuTitle: {
    color: 'white',
    fontSize: 18,
    fontWeight: '600',
    textAlign: 'center'
  },
  qualityOption: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingVertical: 12,
    paddingHorizontal: 16,
    borderRadius: 8,
    marginVertical: 2,
    marginHorizontal: 4
  },
  qualityOptionSelected: {
    backgroundColor: 'rgba(0, 122, 255, 0.2)'
  },
  qualityOptionContent: {
    flex: 1
  },
  qualityOptionText: {
    color: 'white',
    fontSize: 16,
    fontWeight: '500'
  },
  qualityOptionTextSelected: {
    color: '#007AFF',
    fontWeight: '600'
  },
  qualityOptionSubtext: {
    color: 'rgba(255, 255, 255, 0.7)',
    fontSize: 12,
    marginTop: 2
  }
});
