// cloudinary-react-native/lib/module/widgets/video/layer/components/Seekbar.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import React from 'react';
import { View, Text } from 'react-native';
import { TimePosition } from '../types';
import { styles, getResponsiveStyles } from '../styles';
import { SEEKBAR_HEIGHT, COLORS } from '../constants';
export const Seekbar = ({
  progress,
  currentPosition,
  status,
  formatTime,
  seekBarRef,
  panResponder,
  isLandscape = false,
  seekBar = {}
}) => {
  const responsiveStyles = getResponsiveStyles(isLandscape);

  // Get values from config with fallbacks to current defaults
  const seekbarHeight = (seekBar === null || seekBar === void 0 ? void 0 : seekBar.height) ?? SEEKBAR_HEIGHT;
  const seekbarWidth = seekBar === null || seekBar === void 0 ? void 0 : seekBar.width; // undefined means full width (default)
  const seekbarColor = (seekBar === null || seekBar === void 0 ? void 0 : seekBar.color) ?? COLORS.seekbarProgress;
  const timePosition = (seekBar === null || seekBar === void 0 ? void 0 : seekBar.timePosition) ?? TimePosition.BELOW;

  // Create consistent spacing with wrapper View - different heights for above/below
  const timeTextSpacingAbove = isLandscape ? 22 : 24; // Total height for above position
  const timeTextSpacingBelow = isLandscape ? 12 : 14; // Much smaller height for below position

  const timeText = /*#__PURE__*/React.createElement(View, {
    style: {
      height: timePosition === TimePosition.ABOVE ? timeTextSpacingAbove : timeTextSpacingBelow,
      justifyContent: 'center',
      alignItems: 'flex-start',
      paddingLeft: 12
    }
  }, /*#__PURE__*/React.createElement(Text, {
    style: [styles.timeText, {
      lineHeight: 16
    }]
  }, formatTime(currentPosition), " / ", formatTime((status === null || status === void 0 ? void 0 : status.durationMillis) || 0)));

  // Create seekbar width styling
  const seekbarWidthStyle = seekbarWidth ? {
    width: seekbarWidth,
    // Accept both number (px) and string (%)
    alignSelf: 'center' // Center the seekbar if it's not full width
  } : {};
  return /*#__PURE__*/React.createElement(View, {
    style: responsiveStyles.seekbarContainer
  }, /*#__PURE__*/React.createElement(View, {
    style: seekbarWidth ? {
      alignItems: 'center'
    } : {}
  }, timePosition === TimePosition.ABOVE && /*#__PURE__*/React.createElement(View, {
    style: seekbarWidthStyle
  }, timeText), /*#__PURE__*/React.createElement(View, _extends({
    ref: seekBarRef,
    style: [responsiveStyles.seekbar, {
      height: seekbarHeight
    }, seekbarWidthStyle]
  }, panResponder.panHandlers), /*#__PURE__*/React.createElement(View, {
    style: responsiveStyles.seekbarTrack
  }), /*#__PURE__*/React.createElement(View, {
    style: [responsiveStyles.seekbarProgress, {
      width: `${progress * 100}%`,
      backgroundColor: seekbarColor
    }]
  }), /*#__PURE__*/React.createElement(View, {
    style: [responsiveStyles.seekbarHandle, {
      left: `${progress * 100}%`
    }]
  })), timePosition === TimePosition.BELOW && /*#__PURE__*/React.createElement(View, {
    style: seekbarWidthStyle
  }, timeText)));
};
