// cloudinary-react-native/lib/module/widgets/video/layer/components/TitleSubtitle.js
import React from 'react';
import { View, Text } from 'react-native';
import { ButtonPosition } from '../types';
import { getTopPadding } from '../utils';
export function TitleSubtitle({
  isControlsVisible,
  title,
  subtitle,
  isLandscape,
  onBack,
  backButtonPosition,
  titleLeftOffset
}) {
  if (!isControlsVisible || !title && !subtitle) return null;
  return /*#__PURE__*/React.createElement(View, {
    style: [{
      position: 'absolute',
      top: getTopPadding(isLandscape) + (isLandscape ? 6 : 8),
      left: titleLeftOffset !== undefined ? titleLeftOffset : onBack && backButtonPosition === ButtonPosition.NW ? 80 : 20,
      // Custom offset or default positioning
      zIndex: 15,
      maxWidth: '60%' // Prevent overlap with right side buttons
    }]
  }, title && /*#__PURE__*/React.createElement(Text, {
    style: {
      color: 'white',
      fontSize: isLandscape ? 16 : 18,
      fontWeight: 'bold',
      textShadowColor: 'rgba(0,0,0,0.8)',
      textShadowOffset: {
        width: 1,
        height: 1
      },
      textShadowRadius: 2,
      marginBottom: 2
    },
    numberOfLines: 1
  }, title), subtitle && /*#__PURE__*/React.createElement(Text, {
    style: {
      color: 'rgba(255,255,255,0.8)',
      fontSize: isLandscape ? 12 : 14,
      fontWeight: '500',
      textShadowColor: 'rgba(0,0,0,0.8)',
      textShadowOffset: {
        width: 1,
        height: 1
      },
      textShadowRadius: 2
    },
    numberOfLines: 1
  }, subtitle));
}
