// cloudinary-react-native/lib/module/widgets/video/layer/components/TopControls.js
import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { ButtonPosition, ButtonLayoutDirection } from '../types';
import { getResponsiveStyles } from '../styles';
import { ICON_SIZES } from '../constants';
import { CustomButton } from './CustomButton';
export const TopControls = ({
  onBack,
  onShare,
  backButtonPosition,
  shareButtonPosition,
  isLandscape = false,
  fullScreen,
  isFullScreen = false,
  onToggleFullScreen,
  buttonGroups = []
}) => {
  const responsiveStyles = getResponsiveStyles(isLandscape);
  const getPositionStyle = position => {
    switch (position) {
      case ButtonPosition.NE:
        return responsiveStyles.buttonPositionNE;
      case ButtonPosition.NW:
        return responsiveStyles.buttonPositionNW;
      case ButtonPosition.N:
        return responsiveStyles.buttonPositionN;
      case ButtonPosition.SE:
        return responsiveStyles.buttonPositionSE;
      case ButtonPosition.SW:
        return responsiveStyles.buttonPositionSW;
      case ButtonPosition.S:
        return responsiveStyles.buttonPositionS;
      case ButtonPosition.E:
        return responsiveStyles.buttonPositionE;
      case ButtonPosition.W:
        return responsiveStyles.buttonPositionW;
      default:
        return {};
    }
  };

  // Create default full screen button if enabled
  const defaultFullScreenButton = (fullScreen === null || fullScreen === void 0 ? void 0 : fullScreen.enabled) === true && fullScreen !== null && fullScreen !== void 0 && fullScreen.button ? {
    ...fullScreen.button,
    onPress: fullScreen.button.onPress || onToggleFullScreen
  } : (fullScreen === null || fullScreen === void 0 ? void 0 : fullScreen.enabled) === true ? {
    icon: isFullScreen ? 'contract-outline' : 'expand-outline',
    position: ButtonPosition.NE,
    onPress: onToggleFullScreen
  } : null;

  // Process button groups format for top positions
  const processedButtonGroups = {};
  buttonGroups.filter(group => [ButtonPosition.N, ButtonPosition.NE, ButtonPosition.NW].includes(group.position)).forEach(group => {
    processedButtonGroups[group.position] = {
      buttons: group.buttons,
      layoutDirection: group.layoutDirection || ButtonLayoutDirection.VERTICAL
    };
  });

  // Add default full screen button if enabled and not already in a group
  if (defaultFullScreenButton && !processedButtonGroups[ButtonPosition.NE]) {
    processedButtonGroups[ButtonPosition.NE] = {
      buttons: [defaultFullScreenButton],
      layoutDirection: ButtonLayoutDirection.VERTICAL
    };
  } else if (defaultFullScreenButton && processedButtonGroups[ButtonPosition.NE]) {
    // Check if full screen button is already in the group to avoid duplicates
    const existingButtons = processedButtonGroups[ButtonPosition.NE].buttons;
    const hasFullScreenButton = existingButtons.some(button => button.icon === defaultFullScreenButton.icon || button.icon === 'expand-outline' || button.icon === 'contract-outline');
    if (!hasFullScreenButton) {
      processedButtonGroups[ButtonPosition.NE].buttons.push(defaultFullScreenButton);
    }
  }

  // Extract buttons for each position
  const topPositionedButtons = Object.values(processedButtonGroups).flatMap(group => group.buttons);

  // Check if we have any top-positioned buttons (NE, NW, N)
  const hasTopPositionedButtons = backButtonPosition && [ButtonPosition.N, ButtonPosition.NE, ButtonPosition.NW].includes(backButtonPosition) || shareButtonPosition && [ButtonPosition.N, ButtonPosition.NE, ButtonPosition.NW].includes(shareButtonPosition) || topPositionedButtons.length > 0;

  // If we have top-positioned buttons, render them within the bar
  if (hasTopPositionedButtons) {
    var _processedButtonGroup, _processedButtonGroup2, _processedButtonGroup3;
    return /*#__PURE__*/React.createElement(View, {
      style: responsiveStyles.topControlsBar
    }, /*#__PURE__*/React.createElement(View, {
      style: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'flex-start'
      }
    }, onBack && backButtonPosition === ButtonPosition.NW && /*#__PURE__*/React.createElement(TouchableOpacity, {
      style: responsiveStyles.topButton,
      onPress: onBack
    }, /*#__PURE__*/React.createElement(Ionicons, {
      name: "close",
      size: ICON_SIZES.top,
      color: "white"
    })), shareButtonPosition === ButtonPosition.NW && /*#__PURE__*/React.createElement(TouchableOpacity, {
      style: responsiveStyles.topButton,
      onPress: onShare
    }, /*#__PURE__*/React.createElement(Ionicons, {
      name: "share-outline",
      size: ICON_SIZES.top,
      color: "white"
    })), (_processedButtonGroup = processedButtonGroups[ButtonPosition.NW]) === null || _processedButtonGroup === void 0 ? void 0 : _processedButtonGroup.buttons.map((button, index) => /*#__PURE__*/React.createElement(CustomButton, {
      key: `nw-${index}`,
      config: button,
      isLandscape: isLandscape,
      defaultOnPress: button === defaultFullScreenButton ? onToggleFullScreen : undefined
    }))), /*#__PURE__*/React.createElement(View, {
      style: {
        flexDirection: 'row'
      }
    }, onBack && backButtonPosition === ButtonPosition.N && /*#__PURE__*/React.createElement(TouchableOpacity, {
      style: responsiveStyles.topButton,
      onPress: onBack
    }, /*#__PURE__*/React.createElement(Ionicons, {
      name: "close",
      size: ICON_SIZES.top,
      color: "white"
    })), shareButtonPosition === ButtonPosition.N && /*#__PURE__*/React.createElement(TouchableOpacity, {
      style: responsiveStyles.topButton,
      onPress: onShare
    }, /*#__PURE__*/React.createElement(Ionicons, {
      name: "share-outline",
      size: ICON_SIZES.top,
      color: "white"
    })), (_processedButtonGroup2 = processedButtonGroups[ButtonPosition.N]) === null || _processedButtonGroup2 === void 0 ? void 0 : _processedButtonGroup2.buttons.map((button, index) => /*#__PURE__*/React.createElement(CustomButton, {
      key: `n-${index}`,
      config: button,
      isLandscape: isLandscape,
      defaultOnPress: button === defaultFullScreenButton ? onToggleFullScreen : undefined
    }))), /*#__PURE__*/React.createElement(View, {
      style: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'flex-end'
      }
    }, onBack && backButtonPosition === ButtonPosition.NE && /*#__PURE__*/React.createElement(TouchableOpacity, {
      style: responsiveStyles.topButton,
      onPress: onBack
    }, /*#__PURE__*/React.createElement(Ionicons, {
      name: "close",
      size: ICON_SIZES.top,
      color: "white"
    })), shareButtonPosition === ButtonPosition.NE && /*#__PURE__*/React.createElement(TouchableOpacity, {
      style: responsiveStyles.topButton,
      onPress: onShare
    }, /*#__PURE__*/React.createElement(Ionicons, {
      name: "share-outline",
      size: ICON_SIZES.top,
      color: "white"
    })), (_processedButtonGroup3 = processedButtonGroups[ButtonPosition.NE]) === null || _processedButtonGroup3 === void 0 ? void 0 : _processedButtonGroup3.buttons.map((button, index) => /*#__PURE__*/React.createElement(CustomButton, {
      key: `ne-${index}`,
      config: button,
      isLandscape: isLandscape,
      defaultOnPress: button === defaultFullScreenButton ? onToggleFullScreen : undefined
    }))));
  }

  // Default layout (original behavior) - only if no positioning is specified
  if (!backButtonPosition && !shareButtonPosition) {
    return /*#__PURE__*/React.createElement(View, {
      style: responsiveStyles.topControlsBar
    }, onBack && /*#__PURE__*/React.createElement(TouchableOpacity, {
      style: responsiveStyles.topButton,
      onPress: onBack
    }, /*#__PURE__*/React.createElement(Ionicons, {
      name: "close",
      size: ICON_SIZES.top,
      color: "white"
    })), /*#__PURE__*/React.createElement(TouchableOpacity, {
      style: responsiveStyles.topButton,
      onPress: onShare
    }, /*#__PURE__*/React.createElement(Ionicons, {
      name: "share-outline",
      size: ICON_SIZES.top,
      color: "white"
    })));
  }

  // Return empty spacer if only SE buttons are specified
  return /*#__PURE__*/React.createElement(View, {
    style: responsiveStyles.topControlsBar
  });
};
