// cloudinary-react-native/lib/commonjs/widgets/video/layer/helpers/videoControlsManager.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VideoControlsManager = void 0;
var _reactNative = require("react-native");
class VideoControlsManager {
  autoHideTimeoutId = null;
  constructor() {
    this.fadeAnim = new _reactNative.Animated.Value(1);
  }
  getFadeAnim() {
    return this.fadeAnim;
  }
  clearAutoHideTimer = () => {
    if (this.autoHideTimeoutId) {
      clearTimeout(this.autoHideTimeoutId);
      this.autoHideTimeoutId = null;
    }
  };
  startAutoHideTimer = onHide => {
    this.clearAutoHideTimer();
    this.autoHideTimeoutId = setTimeout(() => {
      onHide();
      _reactNative.Animated.timing(this.fadeAnim, {
        toValue: 0,
        duration: 300,
        easing: _reactNative.Easing.linear,
        useNativeDriver: true
      }).start();
    }, 3000);
  };
  toggleControls = (isControlsVisible, onToggle) => {
    const newVisibility = !isControlsVisible;
    onToggle(newVisibility);
    _reactNative.Animated.timing(this.fadeAnim, {
      toValue: newVisibility ? 1 : 0,
      duration: 300,
      easing: _reactNative.Easing.linear,
      useNativeDriver: true
    }).start();
    if (newVisibility) {
      this.startAutoHideTimer(() => onToggle(false));
    } else {
      this.clearAutoHideTimer();
    }
  };
  cleanup = () => {
    this.clearAutoHideTimer();
  };
}
exports.VideoControlsManager = VideoControlsManager;
