// cloudinary-react-native/lib/commonjs/widgets/video/layer/helpers/videoSubtitlesManager.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VideoSubtitlesManager = void 0;
var _subtitleUtils = require("../utils/subtitleUtils");
class VideoSubtitlesManager {
  async handleSubtitleChange(languageCode, state, updateState) {
    updateState({
      currentSubtitle: languageCode,
      activeSubtitleText: null
    });
    if (languageCode === 'off') {
      updateState({
        subtitleCues: [],
        activeSubtitleText: null
      });
      return;
    }
    const selectedTrack = state.availableSubtitleTracks.find(track => track.code === languageCode);
    if (selectedTrack !== null && selectedTrack !== void 0 && selectedTrack.url) {
      try {
        const subtitleCues = await (0, _subtitleUtils.fetchSubtitleFile)(selectedTrack.url);
        updateState({
          subtitleCues
        });
      } catch (error) {
        console.warn('Failed to load subtitle file:', error);
        updateState({
          subtitleCues: []
        });
      }
    } else {
      console.warn('No URL found for subtitle track:', languageCode);
      updateState({
        subtitleCues: []
      });
    }
  }
  handleToggleSubtitlesMenu(isVisible, updateState) {
    updateState({
      isSubtitlesMenuVisible: !isVisible
    });
  }
  updateActiveSubtitle(status, state, updateState) {
    const {
      subtitleCues,
      currentSubtitle
    } = state;
    if (currentSubtitle === 'off' || subtitleCues.length === 0 || !(status !== null && status !== void 0 && status.isLoaded)) {
      if (state.activeSubtitleText !== null) {
        updateState({
          activeSubtitleText: null
        });
      }
      return;
    }
    const currentTimeSeconds = (status.positionMillis || 0) / 1000;
    const activeSubtitle = (0, _subtitleUtils.findActiveSubtitle)(subtitleCues, currentTimeSeconds);
    const newSubtitleText = (activeSubtitle === null || activeSubtitle === void 0 ? void 0 : activeSubtitle.text) || null;
    if (state.activeSubtitleText !== newSubtitleText) {
      updateState({
        activeSubtitleText: newSubtitleText
      });
    }
  }
}
exports.VideoSubtitlesManager = VideoSubtitlesManager;
