// cloudinary-react-native/lib/module/widgets/video/layer/types.js
export let ButtonPosition = /*#__PURE__*/function (ButtonPosition) {
  ButtonPosition["NE"] = "NE";
  // North East (top-right)
  ButtonPosition["NW"] = "NW";
  // North West (top-left)
  ButtonPosition["N"] = "N";
  // North (top-center)
  ButtonPosition["SE"] = "SE";
  // South East (bottom-right)
  ButtonPosition["SW"] = "SW";
  // South West (bottom-left)
  ButtonPosition["S"] = "S";
  // South (bottom-center)
  ButtonPosition["E"] = "E";
  // East (middle-right)
  ButtonPosition["W"] = "W"; // West (middle-left)
  return ButtonPosition;
}({});
export let ButtonLayoutDirection = /*#__PURE__*/function (ButtonLayoutDirection) {
  ButtonLayoutDirection["HORIZONTAL"] = "horizontal";
  // Arrange buttons side by side
  ButtonLayoutDirection["VERTICAL"] = "vertical"; // Arrange buttons stacked vertically
  return ButtonLayoutDirection;
}({});
export let TimePosition = /*#__PURE__*/function (TimePosition) {
  TimePosition["ABOVE"] = "above";
  TimePosition["BELOW"] = "below";
  TimePosition["NONE"] = "none";
  return TimePosition;
}({});
