// cloudinary-react-native/lib/module/widgets/video/layer/helpers/videoOrientationManager.js
import { Dimensions } from 'react-native';
export class VideoOrientationManager {
  orientationSubscription = null;
  orientationCheckInterval = null;
  initialize(updateState) {
    // Subscribe to orientation changes
    this.orientationSubscription = Dimensions.addEventListener('change', ({
      window
    }) => {
      const {
        width,
        height
      } = window;
      const isLandscape = width > height;
      updateState({
        isLandscape
      });
    });

    // Also check orientation periodically as fallback
    this.orientationCheckInterval = setInterval(() => {
      const {
        width,
        height
      } = Dimensions.get('window');
      const isLandscape = width > height;
      // Note: We'd need the current state to compare, but this is handled in the component
      updateState({
        isLandscape
      });
    }, 500);
  }
  cleanup() {
    if (this.orientationSubscription && this.orientationSubscription.remove) {
      this.orientationSubscription.remove();
    }
    if (this.orientationCheckInterval) {
      clearInterval(this.orientationCheckInterval);
      this.orientationCheckInterval = null;
    }
  }
  async handleToggleFullScreen(isFullScreen, isLandscape, fullScreen, updateState) {
    if ((fullScreen === null || fullScreen === void 0 ? void 0 : fullScreen.enabled) !== true) {
      return;
    }
    try {
      if (!isFullScreen) {
        const currentOrientation = isLandscape ? 'landscape' : 'portrait';
        updateState({
          previousOrientation: currentOrientation,
          isFullScreen: true
        });
        if (fullScreen !== null && fullScreen !== void 0 && fullScreen.onEnterFullScreen) {
          fullScreen.onEnterFullScreen();
        }
      } else {
        updateState({
          isFullScreen: false,
          previousOrientation: null
        });
        if (fullScreen !== null && fullScreen !== void 0 && fullScreen.onExitFullScreen) {
          fullScreen.onExitFullScreen();
        }
      }
    } catch (error) {
      console.warn('Failed to toggle full screen:', error);
    }
  }
}
