// cloudinary-react-native/lib/module/widgets/video/analytics/utils/events.js
import { isCustomDataValid, parseCustomData, parseCustomerVideoData, useCustomerVideoDataFallback } from './customer-data';
import { VIEW_EVENT } from '../events.consts';
import { getVideoPlayerType, getVideoPlayerVersion } from './video-player-options';
import { SDKAnalyticsConstants } from '../../../../internal/SDKAnalyticsConstants';
const ANALYTICS_VERSION = JSON.stringify(SDKAnalyticsConstants.sdkSemver);
export const createRegularVideoViewStartEvent = (baseData, customerOptions) => {
  const customData = parseCustomData(customerOptions === null || customerOptions === void 0 ? void 0 : customerOptions.customData);
  const isValidCustomData = isCustomDataValid(customData);
  const customerVideoDataFromFallback = customerOptions !== null && customerOptions !== void 0 && customerOptions.customVideoUrlFallback ? useCustomerVideoDataFallback(baseData.videoUrl, customerOptions.customVideoUrlFallback) : null;
  const customerVideoData = parseCustomerVideoData(customerVideoDataFromFallback);
  return createEvent(VIEW_EVENT.START, {
    ...baseData,
    analyticsModuleVersion: ANALYTICS_VERSION,
    videoPlayer: {
      type: getVideoPlayerType(customerOptions === null || customerOptions === void 0 ? void 0 : customerOptions.videoPlayerType),
      version: getVideoPlayerVersion(customerOptions === null || customerOptions === void 0 ? void 0 : customerOptions.videoPlayerVersion)
    },
    customerData: {
      ...(isValidCustomData ? {
        providedData: customData
      } : {}),
      ...(customerVideoData ? {
        videoData: customerVideoData
      } : {})
    }
  });
};
export const createRegularVideoViewEndEvent = (baseData = {}) => {
  return createEvent(VIEW_EVENT.END, {
    ...baseData
  });
};
export const createLiveStreamViewStartEvent = (baseData, customerOptions) => {
  return createEvent(VIEW_EVENT.START, {
    ...baseData,
    analyticsModuleVersion: ANALYTICS_VERSION,
    videoPlayer: {
      type: getVideoPlayerType(customerOptions === null || customerOptions === void 0 ? void 0 : customerOptions.videoPlayerType),
      version: getVideoPlayerVersion(customerOptions === null || customerOptions === void 0 ? void 0 : customerOptions.videoPlayerVersion)
    }
  });
};
export const createLiveStreamViewEndEvent = (baseData, customerOptions) => {
  return createEvent(VIEW_EVENT.END, {
    ...baseData
  });
};
export const createEvent = (eventName, eventDetails) => ({
  eventName,
  eventTime: Date.now(),
  eventDetails
});
export const prepareEvents = collectedEvents => {
  const events = [...collectedEvents];
  return JSON.stringify(events);
};
