// cloudinary-react-native/lib/commonjs/widgets/video/analytics/cloudinary-analytics.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.connectCloudinaryAnalytics = void 0;
var _isMobile = require("is-mobile");
var _eventsCollector = require("./events-collector");
var _uniqueIds = require("./utils/unique-ids");
var _sendBeaconRequest = require("./utils/send-beacon-request");
var _events = require("./utils/events");
var _validators = require("./utils/validators");
var _nativeHtmlVideoPlayerAdapter = require("./player-adapters/nativeHtmlVideoPlayerAdapter");
var _customerData = require("./utils/customer-data");
var _pageEvents = require("./utils/page-events");
const CLD_ANALYTICS_ENDPOINTS_LIST = {
  production: {
    default: 'https://video-analytics-api.cloudinary.com/v1/video-analytics',
    liveStreams: 'https://video-analytics-api.cloudinary.com/v1/live-streams'
  },
  development: {
    default: 'http://localhost:3001/events',
    liveStreams: 'http://localhost:3001/events'
  }
};
const CLD_ANALYTICS_ENDPOINT = process.env.NODE_ENV === 'development' ? CLD_ANALYTICS_ENDPOINTS_LIST.development : CLD_ANALYTICS_ENDPOINTS_LIST.production;
const connectCloudinaryAnalytics = (videoElement, mainOptions = {}) => {
  (0, _validators.throwErrorIfInvalid)((0, _validators.mainOptionsValidator)(mainOptions), 'Cloudinary video analytics requires proper options object');
  if (!mainOptions.playerAdapter) {
    mainOptions.playerAdapter = (0, _nativeHtmlVideoPlayerAdapter.nativeHtmlVideoPlayerAdapter)(videoElement);
  }
  let videoTrackingSession = null;
  const userId = (0, _uniqueIds.getUserId)();
  const {
    playerAdapter
  } = mainOptions;
  const isMobileDetected = (0, _isMobile.isMobile)({
    tablet: true,
    featureDetect: true
  });
  const createEventsCollector = (0, _eventsCollector.initEventsCollector)(playerAdapter);
  const clearVideoTracking = () => {
    if (videoTrackingSession) {
      videoTrackingSession.clear();
      videoTrackingSession = null;
    }
  };
  const viewId = {
    _value: (0, _uniqueIds.getVideoViewId)(),
    getValue: () => viewId._value,
    regenerateValue: () => {
      viewId._value = (0, _uniqueIds.getVideoViewId)();
      if (videoTrackingSession) {
        videoTrackingSession.viewId = viewId._value;
      }
      return viewId._value;
    }
  };
  const startManualTracking = (metadata, options = {}) => {
    var _videoTrackingSession;
    if (((_videoTrackingSession = videoTrackingSession) === null || _videoTrackingSession === void 0 ? void 0 : _videoTrackingSession.type) === 'auto') {
      throw 'Cloudinary video analytics auto tracking is already connected with this HTML Video Element, to start manual tracking you need to stop auto tracking';
    }
    (0, _validators.throwErrorIfInvalid)((0, _validators.metadataValidator)(metadata), 'Cloudinary video analytics manual tracking called without necessary data');
    (0, _validators.throwErrorIfInvalid)((0, _validators.trackingOptionsValidator)(options), 'Cloudinary video analytics manual tracking called with invalid options');
    clearVideoTracking();
    if (metadata.type === 'live') {
      _startManualTrackingLiveStream(metadata, options);
    } else {
      options.customVideoUrlFallback = options.customVideoUrlFallback || (() => metadata);
      _startManualTrackingRegularVideo(metadata, options);
    }
  };
  const _startManualTrackingRegularVideo = (metadata, options) => {
    const sendData = data => (0, _sendBeaconRequest.sendBeaconRequest)(CLD_ANALYTICS_ENDPOINT.default, data);
    const videoViewEventCollector = createEventsCollector();
    const finishVideoTracking = () => {
      // multiple events can be triggered one by one for specific browsers but we don't have guarantee which ones
      // in this case send data for first event and for rest just skip it to avoid empty payload
      if (videoViewEventCollector.getCollectedEventsCount() > 0) {
        videoViewEventCollector.addEvent((0, _events.createRegularVideoViewEndEvent)());
        const events = (0, _events.prepareEvents)([...videoViewEventCollector.flushEvents()]);
        sendData({
          userId,
          viewId: viewId.getValue(),
          events
        });
      }
    };
    const pageEventsRemoval = (0, _pageEvents.createPageTracker)({
      onPageViewStart: () => {
        viewId.regenerateValue();
        videoViewEventCollector.start(viewId.getValue());
        videoViewEventCollector.addEvent((0, _events.createRegularVideoViewStartEvent)({
          videoUrl: playerAdapter.getCurrentSrc(),
          trackingType: 'manual'
        }, options));
      },
      onPageViewEnd: () => {
        finishVideoTracking();
        videoViewEventCollector.destroy();
      }
    }, isMobileDetected);
    videoTrackingSession = {
      viewId,
      type: 'manual',
      subtype: 'default',
      clear: () => {
        finishVideoTracking();
        videoViewEventCollector.destroy();
        pageEventsRemoval();
      }
    };
  };
  const _startManualTrackingLiveStream = (metadata, options) => {
    const liveStreamData = (0, _customerData.parseCustomerVideoData)(metadata);
    const sendLiveStreamEvent = event => {
      (0, _sendBeaconRequest.sendBeaconRequest)(CLD_ANALYTICS_ENDPOINT.liveStreams, {
        userId,
        viewId: viewId.getValue(),
        events: (0, _events.prepareEvents)([event])
      });
    };
    const pageEventsRemoval = (0, _pageEvents.createPageTracker)({
      onPageViewStart: () => {
        viewId.regenerateValue();
        sendLiveStreamEvent((0, _events.createLiveStreamViewStartEvent)({
          liveStreamData
        }, options));
      },
      onPageViewEnd: () => {
        sendLiveStreamEvent((0, _events.createLiveStreamViewEndEvent)({
          liveStreamData
        }, options));
      }
    }, isMobileDetected);
    videoTrackingSession = {
      viewId: viewId.getValue(),
      type: 'manual',
      subtype: 'live-stream',
      clear: () => {
        pageEventsRemoval();
      }
    };
  };
  const startAutoTracking = (options = {}) => {
    if (videoTrackingSession) {
      throw 'Cloudinary video analytics tracking is already connected with this HTML Video Element';
    }
    (0, _validators.throwErrorIfInvalid)((0, _validators.trackingOptionsValidator)(options), 'Cloudinary video analytics auto tracking called with invalid options');

    // ... existing code ...

    const startAutoTracking = (options = {}) => {
      if (videoTrackingSession) {
        throw 'Cloudinary video analytics tracking is already connected with this Video component';
      }
      (0, _validators.throwErrorIfInvalid)((0, _validators.trackingOptionsValidator)(options), 'Cloudinary video analytics auto tracking called with invalid options');
      const sendData = data => sendRequest(CLD_ANALYTICS_ENDPOINT.default, data);
      const videoViewEventCollector = createEventsCollector();
      let isCollectorActive = false;
      const finishVideoTracking = () => {
        if (isCollectorActive && videoViewEventCollector.getCollectedEventsCount() > 0) {
          videoViewEventCollector.addEvent((0, _events.createRegularVideoViewEndEvent)());
          const events = (0, _events.prepareEvents)([...videoViewEventCollector.flushEvents()]);
          sendData({
            userId,
            viewId: viewId.getValue(),
            events
          });
        }
      };
      const startVideoTracking = () => {
        const sourceUrl = playerAdapter.getCurrentSrc();
        if (!sourceUrl) {
          return null;
        }

        // Don't start if already tracking
        if (videoTrackingSession) {
          return;
        }
        viewId.regenerateValue();
        videoViewEventCollector.start(viewId.getValue());
        isCollectorActive = true;
        videoViewEventCollector.addEvent((0, _events.createRegularVideoViewStartEvent)({
          videoUrl: sourceUrl,
          trackingType: 'auto'
        }, options));
        videoTrackingSession = {
          viewId,
          type: 'auto',
          clear: () => {
            finishVideoTracking();
            if (isCollectorActive) {
              videoViewEventCollector.destroy();
              isCollectorActive = false;
            }
          }
        };
      };
      createAppStateTracker({
        onAppForeground: () => {
          // Only start if not already tracking
          if (!videoTrackingSession) {
            startVideoTracking();
          }
        },
        onAppBackground: () => {
          if (videoTrackingSession) {
            finishVideoTracking();
            if (isCollectorActive) {
              videoViewEventCollector.destroy();
              isCollectorActive = false;
            }
            // Don't clear the session here, just pause tracking
          }
        }
      });

      // Listen for video load events to start tracking
      playerAdapter.onLoadStart(() => !videoTrackingSession && startVideoTracking());
      playerAdapter.onEmptied(() => clearVideoTracking());
    };
    return {
      startManualTracking,
      stopManualTracking: clearVideoTracking,
      startAutoTracking
    };
  };
};
exports.connectCloudinaryAnalytics = connectCloudinaryAnalytics;
