// cloudinary-react-native/lib/commonjs/widgets/video/layer/components/BottomControls.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BottomControls = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _vectorIcons = require("@expo/vector-icons");
var _types = require("../types");
var _styles = require("../styles");
var _constants = require("../constants");
var _Seekbar = require("./Seekbar");
var _PlaybackSpeedButton = require("./PlaybackSpeedButton");
var _SubtitlesButton = require("./SubtitlesButton");
var _QualityButton = require("./QualityButton");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const BottomControls = ({
  status,
  onPlayPause,
  onMuteToggle,
  formatTime,
  getProgress,
  getCurrentPosition,
  seekBarRef,
  panResponder,
  backButtonPosition,
  shareButtonPosition,
  isLandscape = false,
  seekbar = {},
  playbackSpeed,
  currentPlaybackSpeed,
  onPlaybackSpeedChange,
  isSpeedMenuVisible,
  onToggleSpeedMenu,
  subtitles,
  currentSubtitle,
  onSubtitleChange,
  isSubtitlesMenuVisible,
  onToggleSubtitlesMenu,
  quality,
  currentQuality,
  onQualityChange,
  isQualityMenuVisible,
  onToggleQualityMenu
}) => {
  const responsiveStyles = (0, _styles.getResponsiveStyles)(isLandscape);
  const progress = getProgress();
  const currentPosition = getCurrentPosition();

  // Check if we need to leave space for SE positioned buttons
  const hasSEButton = backButtonPosition === _types.ButtonPosition.SE || shareButtonPosition === _types.ButtonPosition.SE;
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: [responsiveStyles.bottomControlsBar, hasSEButton && {
      paddingRight: 80
    } // Create space for SE button
    ]
  }, /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: responsiveStyles.bottomLeftControls
  }, /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
    style: responsiveStyles.playPauseButton,
    onPress: onPlayPause
  }, /*#__PURE__*/_react.default.createElement(_vectorIcons.Ionicons, {
    name: status !== null && status !== void 0 && status.isPlaying ? 'pause' : 'play',
    size: _constants.ICON_SIZES.bottomPlayPause,
    color: "white",
    style: responsiveStyles.playPauseIcon
  })), /*#__PURE__*/_react.default.createElement(_Seekbar.Seekbar, {
    progress: progress,
    currentPosition: currentPosition,
    status: status,
    formatTime: formatTime,
    seekBarRef: seekBarRef,
    panResponder: panResponder,
    isLandscape: isLandscape,
    seekBar: seekbar
  })), /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: responsiveStyles.bottomRightControls
  }, /*#__PURE__*/_react.default.createElement(_QualityButton.QualityButton, {
    quality: quality,
    currentQuality: currentQuality,
    onQualityChange: onQualityChange,
    isLandscape: isLandscape,
    isMenuVisible: isQualityMenuVisible,
    onToggleMenu: onToggleQualityMenu
  }), /*#__PURE__*/_react.default.createElement(_SubtitlesButton.SubtitlesButton, {
    subtitles: subtitles,
    currentSubtitle: currentSubtitle,
    onSubtitleChange: onSubtitleChange,
    isLandscape: isLandscape,
    isMenuVisible: isSubtitlesMenuVisible,
    onToggleMenu: onToggleSubtitlesMenu
  }), /*#__PURE__*/_react.default.createElement(_PlaybackSpeedButton.PlaybackSpeedButton, {
    playbackSpeed: playbackSpeed,
    currentSpeed: currentPlaybackSpeed,
    onSpeedChange: onPlaybackSpeedChange,
    isLandscape: isLandscape,
    isMenuVisible: isSpeedMenuVisible,
    onToggleMenu: onToggleSpeedMenu
  }), /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
    style: responsiveStyles.volumeButton,
    onPress: onMuteToggle
  }, /*#__PURE__*/_react.default.createElement(_vectorIcons.Ionicons, {
    name: status !== null && status !== void 0 && status.isMuted ? 'volume-mute' : 'volume-high',
    size: _constants.ICON_SIZES.bottomVolume,
    color: "white",
    style: responsiveStyles.volumeIcon
  }))));
};
exports.BottomControls = BottomControls;
