// cloudinary-react-native/lib/commonjs/widgets/video/layer/components/CenterControls.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CenterControls = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _vectorIcons = require("@expo/vector-icons");
var _styles = require("../styles");
var _constants = require("../constants");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const CenterControls = ({
  status,
  onPlayPause
}) => {
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: _styles.styles.centerControls
  }, /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
    style: _styles.styles.centerPlayButton,
    onPress: onPlayPause
  }, /*#__PURE__*/_react.default.createElement(_vectorIcons.Ionicons, {
    name: status !== null && status !== void 0 && status.isPlaying ? 'pause' : 'play',
    size: _constants.ICON_SIZES.center,
    color: "#1a1a1a",
    style: {
      marginLeft: status !== null && status !== void 0 && status.isPlaying ? 0 : 3
    } // Slight offset for play icon visual balance
  })));
};
exports.CenterControls = CenterControls;
