// cloudinary-react-native/lib/commonjs/widgets/video/layer/components/CustomButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CustomButton = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _vectorIcons = require("@expo/vector-icons");
var _styles = require("../styles");
var _constants = require("../constants");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const CustomButton = ({
  config,
  isLandscape = false,
  style,
  defaultOnPress
}) => {
  const responsiveStyles = (0, _styles.getResponsiveStyles)(isLandscape);
  const handlePress = () => {
    if (config.onPress) {
      config.onPress();
    } else if (defaultOnPress) {
      defaultOnPress();
    }
  };
  const buttonStyle = [responsiveStyles.topButton, {
    backgroundColor: config.backgroundColor || 'rgba(0, 0, 0, 0.7)'
  }, style];
  return /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
    style: buttonStyle,
    onPress: handlePress
  }, /*#__PURE__*/_react.default.createElement(_vectorIcons.Ionicons, {
    name: config.icon,
    size: config.size || _constants.ICON_SIZES.top,
    color: config.color || 'white'
  }));
};
exports.CustomButton = CustomButton;
