// cloudinary-react-native/lib/commonjs/widgets/video/layer/components/Seekbar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Seekbar = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _types = require("../types");
var _styles = require("../styles");
var _constants = require("../constants");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const Seekbar = ({
  progress,
  currentPosition,
  status,
  formatTime,
  seekBarRef,
  panResponder,
  isLandscape = false,
  seekBar = {}
}) => {
  const responsiveStyles = (0, _styles.getResponsiveStyles)(isLandscape);

  // Get values from config with fallbacks to current defaults
  const seekbarHeight = (seekBar === null || seekBar === void 0 ? void 0 : seekBar.height) ?? _constants.SEEKBAR_HEIGHT;
  const seekbarWidth = seekBar === null || seekBar === void 0 ? void 0 : seekBar.width; // undefined means full width (default)
  const seekbarColor = (seekBar === null || seekBar === void 0 ? void 0 : seekBar.color) ?? _constants.COLORS.seekbarProgress;
  const timePosition = (seekBar === null || seekBar === void 0 ? void 0 : seekBar.timePosition) ?? _types.TimePosition.BELOW;

  // Create consistent spacing with wrapper View - different heights for above/below
  const timeTextSpacingAbove = isLandscape ? 22 : 24; // Total height for above position
  const timeTextSpacingBelow = isLandscape ? 12 : 14; // Much smaller height for below position

  const timeText = /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: {
      height: timePosition === _types.TimePosition.ABOVE ? timeTextSpacingAbove : timeTextSpacingBelow,
      justifyContent: 'center',
      alignItems: 'flex-start',
      paddingLeft: 12
    }
  }, /*#__PURE__*/_react.default.createElement(_reactNative.Text, {
    style: [_styles.styles.timeText, {
      lineHeight: 16
    }]
  }, formatTime(currentPosition), " / ", formatTime((status === null || status === void 0 ? void 0 : status.durationMillis) || 0)));

  // Create seekbar width styling
  const seekbarWidthStyle = seekbarWidth ? {
    width: seekbarWidth,
    // Accept both number (px) and string (%)
    alignSelf: 'center' // Center the seekbar if it's not full width
  } : {};
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: responsiveStyles.seekbarContainer
  }, /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: seekbarWidth ? {
      alignItems: 'center'
    } : {}
  }, timePosition === _types.TimePosition.ABOVE && /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: seekbarWidthStyle
  }, timeText), /*#__PURE__*/_react.default.createElement(_reactNative.View, _extends({
    ref: seekBarRef,
    style: [responsiveStyles.seekbar, {
      height: seekbarHeight
    }, seekbarWidthStyle]
  }, panResponder.panHandlers), /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: responsiveStyles.seekbarTrack
  }), /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: [responsiveStyles.seekbarProgress, {
      width: `${progress * 100}%`,
      backgroundColor: seekbarColor
    }]
  }), /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: [responsiveStyles.seekbarHandle, {
      left: `${progress * 100}%`
    }]
  })), timePosition === _types.TimePosition.BELOW && /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: seekbarWidthStyle
  }, timeText)));
};
exports.Seekbar = Seekbar;
