// cloudinary-react-native/lib/commonjs/widgets/video/layer/components/SubtitleDisplay.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SubtitleDisplay = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const SubtitleDisplay = ({
  text,
  isLandscape = false,
  visible = true
}) => {
  if (!visible || !text) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: [styles.container, isLandscape ? styles.containerLandscape : styles.containerPortrait]
  }, /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: styles.textContainer
  }, /*#__PURE__*/_react.default.createElement(_reactNative.Text, {
    style: styles.subtitleText
  }, text)));
};
exports.SubtitleDisplay = SubtitleDisplay;
const styles = _reactNative.StyleSheet.create({
  container: {
    position: 'absolute',
    left: 20,
    right: 20,
    alignItems: 'center',
    justifyContent: 'center'
  },
  containerPortrait: {
    bottom: 80 // Above bottom controls
  },
  containerLandscape: {
    bottom: 60 // Adjusted for landscape
  },
  textContainer: {
    backgroundColor: 'rgba(0, 0, 0, 0.8)',
    paddingHorizontal: 12,
    paddingVertical: 8,
    borderRadius: 6,
    maxWidth: '90%'
  },
  subtitleText: {
    color: 'white',
    fontSize: 16,
    textAlign: 'center',
    fontWeight: '500',
    lineHeight: 20,
    textShadowColor: 'rgba(0, 0, 0, 0.9)',
    textShadowOffset: {
      width: 1,
      height: 1
    },
    textShadowRadius: 2
  }
});
