// cloudinary-react-native/lib/commonjs/widgets/video/layer/components/SubtitlesButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SubtitlesButton = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _vectorIcons = require("@expo/vector-icons");
var _styles = require("../styles");
var _constants = require("../constants");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
// Remove hardcoded default subtitles - now dynamically loaded from HLS manifest

const SubtitlesButton = ({
  subtitles,
  currentSubtitle = 'off',
  onSubtitleChange,
  isLandscape = false,
  isMenuVisible = false,
  onToggleMenu
}) => {
  var _subtitleOptions$find;
  const responsiveStyles = (0, _styles.getResponsiveStyles)(isLandscape);

  // Don't render if subtitles are not enabled or no options available
  if (!(subtitles !== null && subtitles !== void 0 && subtitles.enabled)) {
    return null;
  }
  const subtitleOptions = subtitles.languages || [];

  // Don't render if no subtitle options available (should not happen with our logic but defensive)
  if (subtitleOptions.length === 0) {
    return null;
  }
  const currentSubtitleLabel = ((_subtitleOptions$find = subtitleOptions.find(option => option.code === currentSubtitle)) === null || _subtitleOptions$find === void 0 ? void 0 : _subtitleOptions$find.label) || 'Off';
  const handleSubtitleSelect = languageCode => {
    onToggleMenu === null || onToggleMenu === void 0 || onToggleMenu(); // Close the menu
    onSubtitleChange === null || onSubtitleChange === void 0 || onSubtitleChange(languageCode);
  };
  const toggleMenu = () => {
    onToggleMenu === null || onToggleMenu === void 0 || onToggleMenu();
  };
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: styles.container
  }, /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
    style: [responsiveStyles.volumeButton, styles.subtitlesButton],
    onPress: toggleMenu
  }, /*#__PURE__*/_react.default.createElement(_vectorIcons.Ionicons, {
    name: "chatbox-outline",
    size: _constants.ICON_SIZES.bottomVolume,
    color: "white",
    style: responsiveStyles.volumeIcon
  })), /*#__PURE__*/_react.default.createElement(_reactNative.Modal, {
    transparent: true,
    visible: isMenuVisible,
    animationType: "fade",
    onRequestClose: onToggleMenu
  }, /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
    style: styles.modalOverlay,
    activeOpacity: 1,
    onPress: onToggleMenu
  }, /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: [styles.subtitlesMenu, isLandscape && styles.subtitlesMenuLandscape]
  }, /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: styles.subtitlesMenuHeader
  }, /*#__PURE__*/_react.default.createElement(_reactNative.Text, {
    style: styles.subtitlesMenuTitle
  }, "Subtitles")), subtitleOptions.map(option => /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
    key: option.code,
    style: [styles.subtitleOption, currentSubtitle === option.code && styles.subtitleOptionSelected],
    onPress: () => handleSubtitleSelect(option.code)
  }, /*#__PURE__*/_react.default.createElement(_reactNative.Text, {
    style: [styles.subtitleOptionText, currentSubtitle === option.code && styles.subtitleOptionTextSelected]
  }, option.label), currentSubtitle === option.code && /*#__PURE__*/_react.default.createElement(_vectorIcons.Ionicons, {
    name: "checkmark",
    size: 20,
    color: "#007AFF"
  })))))));
};
exports.SubtitlesButton = SubtitlesButton;
const styles = _reactNative.StyleSheet.create({
  container: {
    position: 'relative'
  },
  subtitlesButton: {
    // Additional styling for the subtitles button if needed
  },
  modalOverlay: {
    flex: 1,
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
    justifyContent: 'center',
    alignItems: 'center'
  },
  subtitlesMenu: {
    backgroundColor: 'rgba(42, 42, 46, 0.95)',
    borderRadius: 16,
    minWidth: 200,
    maxWidth: 280,
    marginHorizontal: 20,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.25,
    shadowRadius: 10,
    elevation: 5
  },
  subtitlesMenuLandscape: {
    maxWidth: 240
  },
  subtitlesMenuHeader: {
    padding: 16,
    borderBottomWidth: 0.5,
    borderBottomColor: 'rgba(255, 255, 255, 0.2)'
  },
  subtitlesMenuTitle: {
    color: 'white',
    fontSize: 17,
    fontWeight: '600',
    textAlign: 'center'
  },
  subtitleOption: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingHorizontal: 20,
    paddingVertical: 16,
    borderBottomWidth: 0.5,
    borderBottomColor: 'rgba(255, 255, 255, 0.1)'
  },
  subtitleOptionSelected: {
    backgroundColor: 'rgba(0, 122, 255, 0.2)'
  },
  subtitleOptionText: {
    color: 'white',
    fontSize: 17,
    fontWeight: '400'
  },
  subtitleOptionTextSelected: {
    color: '#007AFF',
    fontWeight: '600'
  }
});
