// cloudinary-react-native/lib/commonjs/widgets/video/layer/components/TitleSubtitle.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TitleSubtitle = TitleSubtitle;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _types = require("../types");
var _utils = require("../utils");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function TitleSubtitle({
  isControlsVisible,
  title,
  subtitle,
  isLandscape,
  onBack,
  backButtonPosition,
  titleLeftOffset
}) {
  if (!isControlsVisible || !title && !subtitle) return null;
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: [{
      position: 'absolute',
      top: (0, _utils.getTopPadding)(isLandscape) + (isLandscape ? 6 : 8),
      left: titleLeftOffset !== undefined ? titleLeftOffset : onBack && backButtonPosition === _types.ButtonPosition.NW ? 80 : 20,
      // Custom offset or default positioning
      zIndex: 15,
      maxWidth: '60%' // Prevent overlap with right side buttons
    }]
  }, title && /*#__PURE__*/_react.default.createElement(_reactNative.Text, {
    style: {
      color: 'white',
      fontSize: isLandscape ? 16 : 18,
      fontWeight: 'bold',
      textShadowColor: 'rgba(0,0,0,0.8)',
      textShadowOffset: {
        width: 1,
        height: 1
      },
      textShadowRadius: 2,
      marginBottom: 2
    },
    numberOfLines: 1
  }, title), subtitle && /*#__PURE__*/_react.default.createElement(_reactNative.Text, {
    style: {
      color: 'rgba(255,255,255,0.8)',
      fontSize: isLandscape ? 12 : 14,
      fontWeight: '500',
      textShadowColor: 'rgba(0,0,0,0.8)',
      textShadowOffset: {
        width: 1,
        height: 1
      },
      textShadowRadius: 2
    },
    numberOfLines: 1
  }, subtitle));
}
