// cloudinary-react-native/lib/commonjs/widgets/video/layer/components/TopControls.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TopControls = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _vectorIcons = require("@expo/vector-icons");
var _types = require("../types");
var _styles = require("../styles");
var _constants = require("../constants");
var _CustomButton = require("./CustomButton");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const TopControls = ({
  onBack,
  onShare,
  backButtonPosition,
  shareButtonPosition,
  isLandscape = false,
  fullScreen,
  isFullScreen = false,
  onToggleFullScreen,
  buttonGroups = []
}) => {
  const responsiveStyles = (0, _styles.getResponsiveStyles)(isLandscape);
  const getPositionStyle = position => {
    switch (position) {
      case _types.ButtonPosition.NE:
        return responsiveStyles.buttonPositionNE;
      case _types.ButtonPosition.NW:
        return responsiveStyles.buttonPositionNW;
      case _types.ButtonPosition.N:
        return responsiveStyles.buttonPositionN;
      case _types.ButtonPosition.SE:
        return responsiveStyles.buttonPositionSE;
      case _types.ButtonPosition.SW:
        return responsiveStyles.buttonPositionSW;
      case _types.ButtonPosition.S:
        return responsiveStyles.buttonPositionS;
      case _types.ButtonPosition.E:
        return responsiveStyles.buttonPositionE;
      case _types.ButtonPosition.W:
        return responsiveStyles.buttonPositionW;
      default:
        return {};
    }
  };

  // Create default full screen button if enabled
  const defaultFullScreenButton = (fullScreen === null || fullScreen === void 0 ? void 0 : fullScreen.enabled) === true && fullScreen !== null && fullScreen !== void 0 && fullScreen.button ? {
    ...fullScreen.button,
    onPress: fullScreen.button.onPress || onToggleFullScreen
  } : (fullScreen === null || fullScreen === void 0 ? void 0 : fullScreen.enabled) === true ? {
    icon: isFullScreen ? 'contract-outline' : 'expand-outline',
    position: _types.ButtonPosition.NE,
    onPress: onToggleFullScreen
  } : null;

  // Process button groups format for top positions
  const processedButtonGroups = {};
  buttonGroups.filter(group => [_types.ButtonPosition.N, _types.ButtonPosition.NE, _types.ButtonPosition.NW].includes(group.position)).forEach(group => {
    processedButtonGroups[group.position] = {
      buttons: group.buttons,
      layoutDirection: group.layoutDirection || _types.ButtonLayoutDirection.VERTICAL
    };
  });

  // Add default full screen button if enabled and not already in a group
  if (defaultFullScreenButton && !processedButtonGroups[_types.ButtonPosition.NE]) {
    processedButtonGroups[_types.ButtonPosition.NE] = {
      buttons: [defaultFullScreenButton],
      layoutDirection: _types.ButtonLayoutDirection.VERTICAL
    };
  } else if (defaultFullScreenButton && processedButtonGroups[_types.ButtonPosition.NE]) {
    // Check if full screen button is already in the group to avoid duplicates
    const existingButtons = processedButtonGroups[_types.ButtonPosition.NE].buttons;
    const hasFullScreenButton = existingButtons.some(button => button.icon === defaultFullScreenButton.icon || button.icon === 'expand-outline' || button.icon === 'contract-outline');
    if (!hasFullScreenButton) {
      processedButtonGroups[_types.ButtonPosition.NE].buttons.push(defaultFullScreenButton);
    }
  }

  // Extract buttons for each position
  const topPositionedButtons = Object.values(processedButtonGroups).flatMap(group => group.buttons);

  // Check if we have any top-positioned buttons (NE, NW, N)
  const hasTopPositionedButtons = backButtonPosition && [_types.ButtonPosition.N, _types.ButtonPosition.NE, _types.ButtonPosition.NW].includes(backButtonPosition) || shareButtonPosition && [_types.ButtonPosition.N, _types.ButtonPosition.NE, _types.ButtonPosition.NW].includes(shareButtonPosition) || topPositionedButtons.length > 0;

  // If we have top-positioned buttons, render them within the bar
  if (hasTopPositionedButtons) {
    var _processedButtonGroup, _processedButtonGroup2, _processedButtonGroup3;
    return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
      style: responsiveStyles.topControlsBar
    }, /*#__PURE__*/_react.default.createElement(_reactNative.View, {
      style: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'flex-start'
      }
    }, onBack && backButtonPosition === _types.ButtonPosition.NW && /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
      style: responsiveStyles.topButton,
      onPress: onBack
    }, /*#__PURE__*/_react.default.createElement(_vectorIcons.Ionicons, {
      name: "close",
      size: _constants.ICON_SIZES.top,
      color: "white"
    })), shareButtonPosition === _types.ButtonPosition.NW && /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
      style: responsiveStyles.topButton,
      onPress: onShare
    }, /*#__PURE__*/_react.default.createElement(_vectorIcons.Ionicons, {
      name: "share-outline",
      size: _constants.ICON_SIZES.top,
      color: "white"
    })), (_processedButtonGroup = processedButtonGroups[_types.ButtonPosition.NW]) === null || _processedButtonGroup === void 0 ? void 0 : _processedButtonGroup.buttons.map((button, index) => /*#__PURE__*/_react.default.createElement(_CustomButton.CustomButton, {
      key: `nw-${index}`,
      config: button,
      isLandscape: isLandscape,
      defaultOnPress: button === defaultFullScreenButton ? onToggleFullScreen : undefined
    }))), /*#__PURE__*/_react.default.createElement(_reactNative.View, {
      style: {
        flexDirection: 'row'
      }
    }, onBack && backButtonPosition === _types.ButtonPosition.N && /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
      style: responsiveStyles.topButton,
      onPress: onBack
    }, /*#__PURE__*/_react.default.createElement(_vectorIcons.Ionicons, {
      name: "close",
      size: _constants.ICON_SIZES.top,
      color: "white"
    })), shareButtonPosition === _types.ButtonPosition.N && /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
      style: responsiveStyles.topButton,
      onPress: onShare
    }, /*#__PURE__*/_react.default.createElement(_vectorIcons.Ionicons, {
      name: "share-outline",
      size: _constants.ICON_SIZES.top,
      color: "white"
    })), (_processedButtonGroup2 = processedButtonGroups[_types.ButtonPosition.N]) === null || _processedButtonGroup2 === void 0 ? void 0 : _processedButtonGroup2.buttons.map((button, index) => /*#__PURE__*/_react.default.createElement(_CustomButton.CustomButton, {
      key: `n-${index}`,
      config: button,
      isLandscape: isLandscape,
      defaultOnPress: button === defaultFullScreenButton ? onToggleFullScreen : undefined
    }))), /*#__PURE__*/_react.default.createElement(_reactNative.View, {
      style: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'flex-end'
      }
    }, onBack && backButtonPosition === _types.ButtonPosition.NE && /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
      style: responsiveStyles.topButton,
      onPress: onBack
    }, /*#__PURE__*/_react.default.createElement(_vectorIcons.Ionicons, {
      name: "close",
      size: _constants.ICON_SIZES.top,
      color: "white"
    })), shareButtonPosition === _types.ButtonPosition.NE && /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
      style: responsiveStyles.topButton,
      onPress: onShare
    }, /*#__PURE__*/_react.default.createElement(_vectorIcons.Ionicons, {
      name: "share-outline",
      size: _constants.ICON_SIZES.top,
      color: "white"
    })), (_processedButtonGroup3 = processedButtonGroups[_types.ButtonPosition.NE]) === null || _processedButtonGroup3 === void 0 ? void 0 : _processedButtonGroup3.buttons.map((button, index) => /*#__PURE__*/_react.default.createElement(_CustomButton.CustomButton, {
      key: `ne-${index}`,
      config: button,
      isLandscape: isLandscape,
      defaultOnPress: button === defaultFullScreenButton ? onToggleFullScreen : undefined
    }))));
  }

  // Default layout (original behavior) - only if no positioning is specified
  if (!backButtonPosition && !shareButtonPosition) {
    return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
      style: responsiveStyles.topControlsBar
    }, onBack && /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
      style: responsiveStyles.topButton,
      onPress: onBack
    }, /*#__PURE__*/_react.default.createElement(_vectorIcons.Ionicons, {
      name: "close",
      size: _constants.ICON_SIZES.top,
      color: "white"
    })), /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
      style: responsiveStyles.topButton,
      onPress: onShare
    }, /*#__PURE__*/_react.default.createElement(_vectorIcons.Ionicons, {
      name: "share-outline",
      size: _constants.ICON_SIZES.top,
      color: "white"
    })));
  }

  // Return empty spacer if only SE buttons are specified
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: responsiveStyles.topControlsBar
  });
};
exports.TopControls = TopControls;
