// cloudinary-react-native/lib/commonjs/adapters/ExpoAVVideoAdapter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ExpoAVVideoAdapter = void 0;
var _react = _interopRequireDefault(require("react"));
var _types = require("./types");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
class ExpoAVVideoAdapter {
  expoAVModule = null;
  constructor() {
    this.loadExpoAV();
  }
  loadExpoAV() {
    try {
      this.expoAVModule = require('expo-av');
    } catch (error) {
      this.expoAVModule = null;
    }
  }
  isAvailable() {
    return !!(this.expoAVModule && this.expoAVModule.Video);
  }
  getAdapterName() {
    return _types.VideoPlayerType.EXPO_AV;
  }
  renderVideo(props, ref) {
    if (!this.isAvailable()) {
      throw new Error('expo-av is not available');
    }
    const {
      Video
    } = this.expoAVModule;
    return /*#__PURE__*/_react.default.createElement(Video, {
      ref: videoInstance => {
        if (ref && typeof ref === 'object' && 'current' in ref) {
          ref.current = videoInstance;
        }
      },
      source: {
        uri: props.videoUri
      },
      style: props.style,
      useNativeControls: props.useNativeControls || false,
      shouldPlay: false,
      isLooping: false,
      resizeMode: 'contain',
      onPlaybackStatusUpdate: status => {
        if (props.onPlaybackStatusUpdate) {
          props.onPlaybackStatusUpdate(status);
        }
      },
      onError: error => {
        if (props.onError) {
          props.onError(error);
        }
      },
      onLoad: data => {
        var _props$onLoad;
        (_props$onLoad = props.onLoad) === null || _props$onLoad === void 0 || _props$onLoad.call(props, data);
      },
      onLoadStart: data => {
        var _props$onLoadStart;
        (_props$onLoadStart = props.onLoadStart) === null || _props$onLoadStart === void 0 || _props$onLoadStart.call(props, data);
      }
    });
  }
  processStatusUpdate(videoRef, status, previousStatus) {
    // Import and use the existing expo-av status processor
    try {
      const {
        processExpoAVStatus
      } = require('../widgets/video/analytics/player-adapters/expoAVVideoPlayerAdapter');
      processExpoAVStatus(videoRef, status, previousStatus);
    } catch (error) {
      // Silently fail if analytics adapter is not available
    }
  }
  async getAnalyticsAdapter() {
    try {
      const {
        expoAVVideoPlayerAdapter
      } = await Promise.resolve().then(() => _interopRequireWildcard(require('../widgets/video/analytics/player-adapters/expoAVVideoPlayerAdapter')));
      return expoAVVideoPlayerAdapter;
    } catch (error) {
      return null;
    }
  }
}
exports.ExpoAVVideoAdapter = ExpoAVVideoAdapter;
