// cloudinary-react-native/lib/commonjs/adapters/FallbackVideoAdapter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FallbackVideoAdapter = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _types = require("./types");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
class FallbackVideoAdapter {
  constructor(errorMessage = 'No video player available') {
    this.errorMessage = errorMessage;
  }
  isAvailable() {
    return true; // Fallback is always available
  }
  getAdapterName() {
    return _types.VideoPlayerType.FALLBACK;
  }
  renderVideo(props, _ref) {
    return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
      style: [props.style, {
        backgroundColor: 'red',
        justifyContent: 'center',
        alignItems: 'center'
      }]
    }, /*#__PURE__*/_react.default.createElement(_reactNative.Text, {
      style: {
        color: 'white',
        textAlign: 'center',
        padding: 10
      }
    }, this.errorMessage));
  }
  processStatusUpdate() {
    // No-op for fallback
  }
  async getAnalyticsAdapter() {
    return null;
  }
}
exports.FallbackVideoAdapter = FallbackVideoAdapter;
