// cloudinary-react-native/lib/commonjs/widgets/video/analytics/utils/events.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.prepareEvents = exports.createRegularVideoViewStartEvent = exports.createRegularVideoViewEndEvent = exports.createLiveStreamViewStartEvent = exports.createLiveStreamViewEndEvent = exports.createEvent = void 0;
var _customerData = require("./customer-data");
var _events = require("../events.consts");
var _videoPlayerOptions = require("./video-player-options");
var _SDKAnalyticsConstants = require("../../../../internal/SDKAnalyticsConstants");
const ANALYTICS_VERSION = JSON.stringify(_SDKAnalyticsConstants.SDKAnalyticsConstants.sdkSemver);
const createRegularVideoViewStartEvent = (baseData, customerOptions) => {
  const customData = (0, _customerData.parseCustomData)(customerOptions === null || customerOptions === void 0 ? void 0 : customerOptions.customData);
  const isValidCustomData = (0, _customerData.isCustomDataValid)(customData);
  const customerVideoDataFromFallback = customerOptions !== null && customerOptions !== void 0 && customerOptions.customVideoUrlFallback ? (0, _customerData.useCustomerVideoDataFallback)(baseData.videoUrl, customerOptions.customVideoUrlFallback) : null;
  const customerVideoData = (0, _customerData.parseCustomerVideoData)(customerVideoDataFromFallback);
  return createEvent(_events.VIEW_EVENT.START, {
    ...baseData,
    analyticsModuleVersion: ANALYTICS_VERSION,
    videoPlayer: {
      type: (0, _videoPlayerOptions.getVideoPlayerType)(customerOptions === null || customerOptions === void 0 ? void 0 : customerOptions.videoPlayerType),
      version: (0, _videoPlayerOptions.getVideoPlayerVersion)(customerOptions === null || customerOptions === void 0 ? void 0 : customerOptions.videoPlayerVersion)
    },
    customerData: {
      ...(isValidCustomData ? {
        providedData: customData
      } : {}),
      ...(customerVideoData ? {
        videoData: customerVideoData
      } : {})
    }
  });
};
exports.createRegularVideoViewStartEvent = createRegularVideoViewStartEvent;
const createRegularVideoViewEndEvent = (baseData = {}) => {
  return createEvent(_events.VIEW_EVENT.END, {
    ...baseData
  });
};
exports.createRegularVideoViewEndEvent = createRegularVideoViewEndEvent;
const createLiveStreamViewStartEvent = (baseData, customerOptions) => {
  return createEvent(_events.VIEW_EVENT.START, {
    ...baseData,
    analyticsModuleVersion: ANALYTICS_VERSION,
    videoPlayer: {
      type: (0, _videoPlayerOptions.getVideoPlayerType)(customerOptions === null || customerOptions === void 0 ? void 0 : customerOptions.videoPlayerType),
      version: (0, _videoPlayerOptions.getVideoPlayerVersion)(customerOptions === null || customerOptions === void 0 ? void 0 : customerOptions.videoPlayerVersion)
    }
  });
};
exports.createLiveStreamViewStartEvent = createLiveStreamViewStartEvent;
const createLiveStreamViewEndEvent = (baseData, customerOptions) => {
  return createEvent(_events.VIEW_EVENT.END, {
    ...baseData
  });
};
exports.createLiveStreamViewEndEvent = createLiveStreamViewEndEvent;
const createEvent = (eventName, eventDetails) => ({
  eventName,
  eventTime: Date.now(),
  eventDetails
});
exports.createEvent = createEvent;
const prepareEvents = collectedEvents => {
  const events = [...collectedEvents];
  return JSON.stringify(events);
};
exports.prepareEvents = prepareEvents;
