// cloudinary-react-native/lib/module/adapters/ExpoAVVideoAdapter.js
import React from 'react';
import { VideoPlayerType } from './types';
export class ExpoAVVideoAdapter {
  expoAVModule = null;
  constructor() {
    this.loadExpoAV();
  }
  loadExpoAV() {
    try {
      this.expoAVModule = require('expo-av');
    } catch (error) {
      this.expoAVModule = null;
    }
  }
  isAvailable() {
    return !!(this.expoAVModule && this.expoAVModule.Video);
  }
  getAdapterName() {
    return VideoPlayerType.EXPO_AV;
  }
  renderVideo(props, ref) {
    if (!this.isAvailable()) {
      throw new Error('expo-av is not available');
    }
    const {
      Video
    } = this.expoAVModule;
    return /*#__PURE__*/React.createElement(Video, {
      ref: videoInstance => {
        if (ref && typeof ref === 'object' && 'current' in ref) {
          ref.current = videoInstance;
        }
      },
      source: {
        uri: props.videoUri
      },
      style: props.style,
      useNativeControls: props.useNativeControls || false,
      shouldPlay: false,
      isLooping: false,
      resizeMode: 'contain',
      onPlaybackStatusUpdate: status => {
        if (props.onPlaybackStatusUpdate) {
          props.onPlaybackStatusUpdate(status);
        }
      },
      onError: error => {
        if (props.onError) {
          props.onError(error);
        }
      },
      onLoad: data => {
        var _props$onLoad;
        (_props$onLoad = props.onLoad) === null || _props$onLoad === void 0 || _props$onLoad.call(props, data);
      },
      onLoadStart: data => {
        var _props$onLoadStart;
        (_props$onLoadStart = props.onLoadStart) === null || _props$onLoadStart === void 0 || _props$onLoadStart.call(props, data);
      }
    });
  }
  processStatusUpdate(videoRef, status, previousStatus) {
    // Import and use the existing expo-av status processor
    try {
      const {
        processExpoAVStatus
      } = require('../widgets/video/analytics/player-adapters/expoAVVideoPlayerAdapter');
      processExpoAVStatus(videoRef, status, previousStatus);
    } catch (error) {
      // Silently fail if analytics adapter is not available
    }
  }
  async getAnalyticsAdapter() {
    try {
      const {
        expoAVVideoPlayerAdapter
      } = await import('../widgets/video/analytics/player-adapters/expoAVVideoPlayerAdapter');
      return expoAVVideoPlayerAdapter;
    } catch (error) {
      return null;
    }
  }
}
