// cloudinary-react-native/lib/module/adapters/VideoPlayerFactory.js
import { ExpoVideoAdapter } from './ExpoVideoAdapter';
import { ExpoAVVideoAdapter } from './ExpoAVVideoAdapter';
import { FallbackVideoAdapter } from './FallbackVideoAdapter';

// Re-export types for external use
export { VideoPlayerType } from './types';
export class VideoPlayerFactory {
  static adapters = [];
  static initialized = false;
  static initializeAdapters() {
    if (this.initialized) return;
    this.adapters = [new ExpoVideoAdapter(),
    // Try expo-video first (modern, recommended)
    new ExpoAVVideoAdapter() // Fallback to expo-av for compatibility
    ];
    this.initialized = true;
  }

  /**
   * Get the first available video player adapter
   */
  static getAvailableAdapter() {
    this.initializeAdapters();

    // Find the first available adapter
    for (const adapter of this.adapters) {
      const isAvailable = adapter.isAvailable();
      if (isAvailable) {
        return adapter;
      }
    }

    // If no adapter is available, return fallback
    return new FallbackVideoAdapter('No video player library found. Install expo-video or expo-av.');
  }

  /**
   * Get a specific adapter by type
   */
  static getAdapterByType(type) {
    this.initializeAdapters();
    const adapter = this.adapters.find(adapter => adapter.getAdapterName() === type);
    if (adapter && adapter.isAvailable()) {
      return adapter;
    }
    throw new Error(`Adapter for ${type} is not available`);
  }

  /**
   * Get information about all available adapters
   */
  static getAvailableAdapters() {
    this.initializeAdapters();
    return this.adapters.map(adapter => ({
      name: adapter.getAdapterName(),
      available: adapter.isAvailable()
    }));
  }

  /**
   * Check if a specific video player type is available
   */
  static isAdapterAvailable(type) {
    this.initializeAdapters();
    const adapter = this.adapters.find(adapter => adapter.getAdapterName() === type);
    return adapter ? adapter.isAvailable() : false;
  }
}
