// cloudinary-react-native/lib/commonjs/widgets/video/layer/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TOP_PADDING_IOS_LANDSCAPE = exports.TOP_PADDING_IOS = exports.TOP_PADDING_ANDROID_LANDSCAPE = exports.TOP_PADDING_ANDROID = exports.TOP_BUTTON_SIZE = exports.SHADOW_VALUES = exports.SE_BUTTON_RIGHT_OFFSET = exports.SE_BUTTON_BOTTOM_OFFSET_LANDSCAPE = exports.SE_BUTTON_BOTTOM_OFFSET = exports.SEEK_POSITION_TOLERANCE = exports.SEEK_POSITION_THRESHOLD = exports.SEEK_DEBOUNCE_DELAY = exports.SEEK_BUFFER_MS = exports.SEEKBAR_TRACK_HEIGHT = exports.SEEKBAR_HEIGHT = exports.SEEKBAR_HANDLE_SIZE = exports.SEEKBAR_ALIGNMENT_OFFSET_LANDSCAPE = exports.SEEKBAR_ALIGNMENT_OFFSET = exports.LEGACY_TOP_PADDING_IOS = exports.LEGACY_TOP_PADDING_ANDROID = exports.ICON_SIZES = exports.CONTROLS_FADE_DURATION = exports.CONTROLS_AUTO_HIDE_DELAY = exports.COLORS = exports.CENTER_PLAY_BUTTON_SIZE = exports.BUTTON_SPACING = exports.BUTTON_MARGIN = exports.BOTTOM_CONTROLS_PADDING_LANDSCAPE = exports.BOTTOM_CONTROLS_PADDING = exports.BOTTOM_BUTTON_SIZE = exports.BORDER_RADIUS = void 0;
var _reactNative = require("react-native");
// Animation and timing constants
const CONTROLS_AUTO_HIDE_DELAY = exports.CONTROLS_AUTO_HIDE_DELAY = 3000; // 3 seconds
const CONTROLS_FADE_DURATION = exports.CONTROLS_FADE_DURATION = 300; // 300ms
const SEEK_DEBOUNCE_DELAY = exports.SEEK_DEBOUNCE_DELAY = 200; // 200ms
const SEEK_POSITION_THRESHOLD = exports.SEEK_POSITION_THRESHOLD = 100; // 100ms
const SEEK_POSITION_TOLERANCE = exports.SEEK_POSITION_TOLERANCE = 500; // 500ms
const SEEK_BUFFER_MS = exports.SEEK_BUFFER_MS = 100; // 100ms buffer from end

// Responsive UI dimensions - now orientation-aware
const TOP_BUTTON_SIZE = exports.TOP_BUTTON_SIZE = _reactNative.Platform.select({
  ios: 44,
  android: 48
});
const CENTER_PLAY_BUTTON_SIZE = exports.CENTER_PLAY_BUTTON_SIZE = 72; // Center button size
const BOTTOM_BUTTON_SIZE = exports.BOTTOM_BUTTON_SIZE = _reactNative.Platform.select({
  ios: 40,
  android: 44
});
const SEEKBAR_HEIGHT = exports.SEEKBAR_HEIGHT = 20;
const SEEKBAR_TRACK_HEIGHT = exports.SEEKBAR_TRACK_HEIGHT = 3;
const SEEKBAR_HANDLE_SIZE = exports.SEEKBAR_HANDLE_SIZE = 16;

// Platform-specific padding with safe area consideration - orientation-responsive
const TOP_PADDING_IOS = exports.TOP_PADDING_IOS = 60;
const TOP_PADDING_ANDROID = exports.TOP_PADDING_ANDROID = 30;
const LEGACY_TOP_PADDING_IOS = exports.LEGACY_TOP_PADDING_IOS = 50;
const LEGACY_TOP_PADDING_ANDROID = exports.LEGACY_TOP_PADDING_ANDROID = 20;

// Landscape-specific padding (reduced for landscape mode)
const TOP_PADDING_IOS_LANDSCAPE = exports.TOP_PADDING_IOS_LANDSCAPE = 20;
const TOP_PADDING_ANDROID_LANDSCAPE = exports.TOP_PADDING_ANDROID_LANDSCAPE = 6;

// Bottom controls alignment constants (responsive and orientation-aware)
const BOTTOM_CONTROLS_PADDING = exports.BOTTOM_CONTROLS_PADDING = _reactNative.Platform.select({
  ios: 15,
  android: 12
});
const BOTTOM_CONTROLS_PADDING_LANDSCAPE = exports.BOTTOM_CONTROLS_PADDING_LANDSCAPE = _reactNative.Platform.select({
  ios: 10,
  android: 8
});
const SEEKBAR_ALIGNMENT_OFFSET = exports.SEEKBAR_ALIGNMENT_OFFSET = _reactNative.Platform.select({
  ios: 16,
  android: 14
}); // Platform-specific alignment
const SEEKBAR_ALIGNMENT_OFFSET_LANDSCAPE = exports.SEEKBAR_ALIGNMENT_OFFSET_LANDSCAPE = _reactNative.Platform.select({
  ios: 12,
  android: 10
});
// Align SE buttons with volume button position
// Volume button position = bottomControlsBar paddingHorizontal (20) + volumeButton marginRight (8) = 28
const SE_BUTTON_RIGHT_OFFSET = exports.SE_BUTTON_RIGHT_OFFSET = 28; // Align with volume button
const SE_BUTTON_BOTTOM_OFFSET = exports.SE_BUTTON_BOTTOM_OFFSET = _reactNative.Platform.select({
  ios: 32,
  android: 28
}); // Platform-specific bottom offset
const SE_BUTTON_BOTTOM_OFFSET_LANDSCAPE = exports.SE_BUTTON_BOTTOM_OFFSET_LANDSCAPE = _reactNative.Platform.select({
  ios: 20,
  android: 18
});

// Button spacing constants for multiple buttons in the same position
const BUTTON_SPACING = exports.BUTTON_SPACING = _reactNative.Platform.select({
  ios: 8,
  android: 6
}); // Space between buttons
const BUTTON_MARGIN = exports.BUTTON_MARGIN = _reactNative.Platform.select({
  ios: 4,
  android: 3
}); // Margin from edge

// Visual styling constants
const BORDER_RADIUS = exports.BORDER_RADIUS = {
  topButton: 22,
  centerButton: 36,
  bottomButton: 20,
  seekbar: 10,
  seekbarTrack: 1.5,
  seekbarHandle: 8
};

// Shadow and elevation values
const SHADOW_VALUES = exports.SHADOW_VALUES = {
  topButton: {
    shadowOffset: {
      width: 0,
      height: 1
    },
    shadowOpacity: 0.3,
    shadowRadius: 2,
    elevation: 2
  },
  centerButton: {
    shadowOffset: {
      width: 0,
      height: 4
    },
    shadowOpacity: 0.25,
    shadowRadius: 8,
    elevation: 8
  },
  seekbarProgress: {
    shadowOffset: {
      width: 0,
      height: 0
    },
    shadowOpacity: 0.6,
    shadowRadius: 2,
    elevation: 2
  },
  seekbarHandle: {
    shadowOffset: {
      width: 0,
      height: 3
    },
    shadowOpacity: 0.3,
    shadowRadius: 6,
    elevation: 6
  }
};

// Color constants
const COLORS = exports.COLORS = {
  overlay: 'rgba(0, 0, 0, 0.4)',
  loadingBackground: 'rgba(0, 0, 0, 0.8)',
  topControlsBackground: 'rgba(0, 0, 0, 0.6)',
  bottomControlsBackground: 'rgba(0, 0, 0, 0.6)',
  topButtonBackground: 'rgba(0, 0, 0, 0.7)',
  centerButtonBackground: 'rgba(255, 255, 255, 0.95)',
  centerButtonBorder: 'rgba(255, 255, 255, 0.8)',
  seekbarTrack: 'rgba(255, 255, 255, 0.2)',
  seekbarProgress: '#FFFFFF',
  seekbarHandle: '#FFFFFF',
  seekbarHandleBorder: 'rgba(255, 255, 255, 0.9)',
  text: {
    white: 'white',
    black: '#1a1a1a',
    shadow: 'rgba(0, 0, 0, 0.5)',
    whiteShadow: 'rgba(255, 255, 255, 0.3)'
  }
};

// Icon sizes
const ICON_SIZES = exports.ICON_SIZES = {
  top: 24,
  center: 32,
  bottom: 26,
  bottomVolume: 26,
  bottomPlayPause: 26
};
