// cloudinary-react-native/lib/commonjs/widgets/video/layer/styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.styles = exports.getResponsiveStyles = void 0;
var _reactNative = require("react-native");
var _constants = require("./constants");
var _utils = require("./utils");
// Base styles (orientation-independent)
const styles = exports.styles = _reactNative.StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'black'
  },
  overlay: {
    ..._reactNative.StyleSheet.absoluteFillObject,
    justifyContent: 'space-between'
  },
  // Loading styles
  loadingContainer: {
    ..._reactNative.StyleSheet.absoluteFillObject,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: _constants.COLORS.loadingBackground,
    zIndex: 1
  },
  loadingText: {
    color: _constants.COLORS.text.white,
    fontSize: 16,
    marginTop: 16,
    textAlign: 'center',
    opacity: 0.9
  },
  // Center Controls
  centerControls: {
    alignSelf: 'center',
    justifyContent: 'center',
    alignItems: 'center'
  },
  centerPlayButton: {
    width: _constants.CENTER_PLAY_BUTTON_SIZE,
    height: _constants.CENTER_PLAY_BUTTON_SIZE,
    borderRadius: _constants.BORDER_RADIUS.centerButton,
    backgroundColor: _constants.COLORS.centerButtonBackground,
    justifyContent: 'center',
    alignItems: 'center',
    shadowColor: '#000',
    ..._constants.SHADOW_VALUES.centerButton,
    borderWidth: 1,
    borderColor: _constants.COLORS.centerButtonBorder
  },
  centerPlayIcon: {
    color: _constants.COLORS.text.black,
    fontSize: 32,
    fontWeight: '500',
    marginLeft: 2,
    // Slight offset for play icon visual balance
    textShadowColor: _constants.COLORS.text.whiteShadow,
    textShadowOffset: {
      width: 0,
      height: 1
    },
    textShadowRadius: 1
  },
  iconText: {
    color: _constants.COLORS.text.white,
    fontSize: 20,
    fontWeight: '400',
    textShadowColor: _constants.COLORS.text.shadow,
    textShadowOffset: {
      width: 0,
      height: 1
    },
    textShadowRadius: 1
  },
  timeText: {
    color: _constants.COLORS.text.white,
    fontSize: 12,
    opacity: 0.8
  },
  // Legacy styles (keeping for backward compatibility)
  topRow: {
    marginTop: _reactNative.Platform.OS === 'ios' ? _constants.LEGACY_TOP_PADDING_IOS : _constants.LEGACY_TOP_PADDING_ANDROID,
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: 16
  },
  bottomRow: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    padding: 16
  }
});

// Orientation-responsive style functions
const getResponsiveStyles = (isLandscape = false) => {
  const topPadding = (0, _utils.getTopPadding)(isLandscape);
  const bottomPadding = (0, _utils.getBottomControlsPadding)(isLandscape);
  const seekbarOffset = (0, _utils.getSeekbarAlignmentOffset)(isLandscape);
  const seButtonBottomOffset = (0, _utils.getSEButtonBottomOffset)(isLandscape);
  return _reactNative.StyleSheet.create({
    // Top Controls
    topControlsBar: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
      paddingHorizontal: 20,
      paddingTop: topPadding,
      paddingBottom: (bottomPadding || 15) * 2,
      backgroundColor: _constants.COLORS.topControlsBackground
    },
    topButton: {
      width: _constants.TOP_BUTTON_SIZE,
      height: _constants.TOP_BUTTON_SIZE,
      borderRadius: _constants.BORDER_RADIUS.topButton,
      backgroundColor: _constants.COLORS.topButtonBackground,
      justifyContent: 'center',
      alignItems: 'center',
      shadowColor: '#000',
      ..._constants.SHADOW_VALUES.topButton
    },
    // Button positioning styles
    buttonPositionNE: {
      position: 'absolute',
      top: topPadding + (isLandscape ? 6 : 8),
      right: 20
    },
    buttonPositionNW: {
      position: 'absolute',
      top: topPadding + (isLandscape ? 6 : 8),
      left: 20
    },
    buttonPositionN: {
      position: 'absolute',
      top: topPadding + (isLandscape ? 6 : 8),
      alignSelf: 'center'
    },
    buttonPositionSE: {
      position: 'absolute',
      bottom: seButtonBottomOffset,
      right: _constants.SE_BUTTON_RIGHT_OFFSET,
      zIndex: 10
    },
    buttonPositionSW: {
      position: 'absolute',
      bottom: seButtonBottomOffset,
      left: _constants.SE_BUTTON_RIGHT_OFFSET,
      zIndex: 10
    },
    buttonPositionS: {
      position: 'absolute',
      bottom: seButtonBottomOffset,
      alignSelf: 'center',
      zIndex: 10
    },
    buttonPositionE: {
      position: 'absolute',
      right: 20,
      top: '50%',
      marginTop: -22,
      // Half of TOP_BUTTON_SIZE to center
      zIndex: 10
    },
    buttonPositionW: {
      position: 'absolute',
      left: 20,
      top: '50%',
      marginTop: -22,
      // Half of TOP_BUTTON_SIZE to center
      zIndex: 10
    },
    // Bottom Controls
    bottomControlsBar: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
      paddingHorizontal: 20,
      paddingVertical: bottomPadding,
      backgroundColor: _constants.COLORS.bottomControlsBackground,
      zIndex: 10 // Ensure bottom controls appear above button bar
    },
    bottomLeftControls: {
      flex: 1,
      flexDirection: 'row',
      alignItems: 'center'
    },
    bottomRightControls: {
      flexDirection: 'row',
      alignItems: 'center'
    },
    playPauseButton: {
      width: _constants.BOTTOM_BUTTON_SIZE,
      height: _constants.BOTTOM_BUTTON_SIZE,
      borderRadius: _constants.BORDER_RADIUS.bottomButton,
      backgroundColor: 'transparent',
      justifyContent: 'center',
      alignItems: 'center',
      marginRight: 12,
      marginBottom: seekbarOffset
    },
    playPauseIcon: {
      color: _constants.COLORS.text.white,
      fontSize: isLandscape ? 20 : 22,
      fontWeight: '500',
      textShadowColor: _constants.COLORS.text.shadow,
      textShadowOffset: {
        width: 0,
        height: 1
      },
      textShadowRadius: 1
    },
    volumeButton: {
      width: _constants.BOTTOM_BUTTON_SIZE,
      height: _constants.BOTTOM_BUTTON_SIZE,
      borderRadius: _constants.BORDER_RADIUS.bottomButton,
      backgroundColor: 'transparent',
      justifyContent: 'center',
      alignItems: 'center',
      marginRight: 8,
      marginBottom: seekbarOffset
    },
    volumeIcon: {
      color: _constants.COLORS.text.white,
      fontSize: isLandscape ? 18 : 20,
      fontWeight: '500',
      textShadowColor: _constants.COLORS.text.shadow,
      textShadowOffset: {
        width: 0,
        height: 1
      },
      textShadowRadius: 1
    },
    // Seekbar
    seekbarContainer: {
      flex: 1,
      marginRight: 18,
      marginLeft: 8
    },
    seekbar: {
      height: _constants.SEEKBAR_HEIGHT,
      borderRadius: _constants.BORDER_RADIUS.seekbar,
      position: 'relative',
      justifyContent: 'center',
      paddingVertical: isLandscape ? 6 : 8 // Reduce touch area in landscape
    },
    seekbarTrack: {
      height: _constants.SEEKBAR_TRACK_HEIGHT,
      backgroundColor: _constants.COLORS.seekbarTrack,
      borderRadius: _constants.BORDER_RADIUS.seekbarTrack,
      position: 'absolute',
      top: 8.5,
      left: 0,
      right: 0
    },
    seekbarProgress: {
      height: _constants.SEEKBAR_TRACK_HEIGHT,
      backgroundColor: _constants.COLORS.seekbarProgress,
      borderRadius: _constants.BORDER_RADIUS.seekbarTrack,
      position: 'absolute',
      top: 8.5,
      // Center within the 20px height
      shadowColor: _constants.COLORS.seekbarProgress,
      ..._constants.SHADOW_VALUES.seekbarProgress
    },
    seekbarHandle: {
      position: 'absolute',
      width: _constants.SEEKBAR_HANDLE_SIZE,
      height: _constants.SEEKBAR_HANDLE_SIZE,
      borderRadius: _constants.BORDER_RADIUS.seekbarHandle,
      backgroundColor: _constants.COLORS.seekbarHandle,
      top: 2,
      // Center within the 20px height
      marginLeft: -8,
      // Half of width to center properly
      shadowColor: '#000',
      ..._constants.SHADOW_VALUES.seekbarHandle,
      borderWidth: 2,
      borderColor: _constants.COLORS.seekbarHandleBorder
    }
  });
};
exports.getResponsiveStyles = getResponsiveStyles;
