// cloudinary-react-native/lib/commonjs/widgets/video/layer/types.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TimePosition = exports.ButtonPosition = exports.ButtonLayoutDirection = void 0;
let ButtonPosition = exports.ButtonPosition = /*#__PURE__*/function (ButtonPosition) {
  ButtonPosition["NE"] = "NE";
  // North East (top-right)
  ButtonPosition["NW"] = "NW";
  // North West (top-left)
  ButtonPosition["N"] = "N";
  // North (top-center)
  ButtonPosition["SE"] = "SE";
  // South East (bottom-right)
  ButtonPosition["SW"] = "SW";
  // South West (bottom-left)
  ButtonPosition["S"] = "S";
  // South (bottom-center)
  ButtonPosition["E"] = "E";
  // East (middle-right)
  ButtonPosition["W"] = "W"; // West (middle-left)
  return ButtonPosition;
}({});
let ButtonLayoutDirection = exports.ButtonLayoutDirection = /*#__PURE__*/function (ButtonLayoutDirection) {
  ButtonLayoutDirection["HORIZONTAL"] = "horizontal";
  // Arrange buttons side by side
  ButtonLayoutDirection["VERTICAL"] = "vertical"; // Arrange buttons stacked vertically
  return ButtonLayoutDirection;
}({});
let TimePosition = exports.TimePosition = /*#__PURE__*/function (TimePosition) {
  TimePosition["ABOVE"] = "above";
  TimePosition["BELOW"] = "below";
  TimePosition["NONE"] = "none";
  return TimePosition;
}({});
